metashare/api/jobs.py

```diff
@@ -470,3 +471,39 @@ def populate_github_users(repository):
 
 
 populate_github_users_job = job(populate_github_users)
+
+
+def submit_review(*, user, task, data):
+    try:
+        task.refresh_from_db()
+        repository = get_repo_info(user, repo_id=task.project.repository.repo_id)
+        pr = repository.pull_request(task.pr_number)
+        # We always COMMENT so as not to change the PR's status:
+        pr.create_review(data["notes"], event="COMMENT")
+        if task.commits:
+            state_for_status = {
+                # "": "pending",
+                # "": "error",
+                TASK_REVIEW_STATUS.Approved: "success",
+                TASK_REVIEW_STATUS["Changes requested"]: "failure",
+            }.get(data["status"])
+            repository.create_status(
+                task.commits[0],
```

Comment:
@wlonk I'm trying to step back a bit and think more carefully about the architecture here. I think this would technically work, since on the front end we only allow submitting a review when the org is up-to-date. But it also feels a bit weird to always assume the review applies to the most recent commit we know about. And we also have the problem of the "Delete org" option...

So here's a slightly modified proposal for your consideration:

- We add a `task.review_sha` field which stores the sha that the review was submitted on.
- We leave the logic attached to a `task`, but the view/serializer also accepts an _optional_ `org` option, which is a `scratch_org.id`.
- If `org` is included, we create the status on `org.latest_commit`, and we can potentially act on the `delete_org_on_submit` flag by deleting the org with that id.
- If `org` is not included and `task.review_valid` is truthy, we create the status on `task.review_sha`, and we ignore `delete_org_on_submit`.

Reply:
Talked through this.

We want the frontend to submit the org id in question, and the backend to validate that that's a review org for this task. Trust, but verify. This lets us do many possibly useful things in future, too.

Otherwise, all the above.

Reply:
@wlonk This was not working functionally -- I made a few changes in https://github.com/oddbird/MetaShare/pull/164/commits/160dd62934734faebc7518ca57a754e772ad9673 for you to review.